### PR TITLE
[1.0-dev] Add an example where we toggle between view and edit

### PR DESCRIPTION
**Pull request by cgestes**
It helps catching bugs when we change Zefyr properties, I believe this one is a common usecase.

It is an example of a readonly view.

### CODE REVIEW

File: packages/zefyr/example/lib/src/home.dart [hunk 1]
```diff
@@ -129,6 +130,12 @@ class _HomePageState extends State<HomePage> {
           visualDensity: VisualDensity.compact,
           onTap: _scrollable,
         ),
+        ListTile(
+          title: Text('¶   Edit button', style: itemStyle),
```
Comment (pulyaevskiy) [inline packages/zefyr/example/lib/src/home.dart:134]: Let's call it "Read only view"?

Comment (pulyaevskiy) [reply packages/zefyr/example/lib/src/home.dart:134]: Also, create a new section "BASICS" in this menu and probably make it the top one. Move this example to the BASICS section, of course.

File: packages/zefyr/example/lib/src/edit_button_layout.dart [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+class _EditButtonLayoutState extends State<EditButtonLayout> {
+  final FocusNode _focusNode = FocusNode();
+
+  bool _edit = false;
+
+  @override
+  Widget build(BuildContext context) {
+    return DemoScaffold(
+      documentFilename: 'layout_expanded.note',
```
Comment (pulyaevskiy) [inline packages/zefyr/example/lib/src/edit_button_layout.dart:19]: This needs to be specific to the example, e.g. "basics_read_only_view.note".

You don't have to create the actual document it'll default to an "Empty asset". But if you feel adventurous you can create and edit it using following steps:

1. Run the app in desktop mode.
2. On the home screen click on the "cog" icon next to Zefyr title
3. Paste path to the assets folder. E.g. `/path/to/zefyr/packages/zefyr/example/assets`.
4. Save the dialog
5. Open the new example, you'll see default content in the editor "Empty asset"
6. Type in whatever example description you think would be appropriate.
7. Click on the "Save" button in the top-right corner
8. Done

File: packages/zefyr/example/lib/src/edit_button_layout.dart [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    return ZefyrEditor(
+      controller: controller,
+      focusNode: _focusNode,
+      expands: true,
+      readOnly: true,
+      showCursor: false,
+      padding: EdgeInsets.symmetric(horizontal: 8),
+    );
```
Comment (pulyaevskiy) [inline packages/zefyr/example/lib/src/edit_button_layout.dart:47]: Can simplify this to:

```dart
    return ZefyrEditor(
      controller: controller,
      focusNode: _focusNode,
      expands: true,
      readOnly: !_edit,
      showCursor: _edit,
      padding: EdgeInsets.symmetric(horizontal: 8),
    );
```

File: packages/zefyr/example/lib/src/forms_decorated_field.dart [hunk 4]
```diff
@@ -16,6 +16,7 @@ class _DecoratedFieldDemoState extends State<DecoratedFieldDemo> {
     return DemoScaffold(
       documentFilename: 'decorated_field.note',
       builder: _buildContent,
+      showToolbar: false,
```
Comment (pulyaevskiy) [inline packages/zefyr/example/lib/src/forms_decorated_field.dart:19]: Thanks!

**Review by pulyaevskiy — CHANGES_REQUESTED** [review]
Thanks, this is nice! Have some comments around the structure and contents, but otherwise looks good.

Please take a look.

**Comment by cgestes** [comment]
Will makes the changes :)

**Comment by cgestes** [comment]
Should be fine now. Tell me if it needs more changes.

PS: Wow Linux support is working really well :+1:

**Review by pulyaevskiy — APPROVED** [review]
LGTM, thanks!

**Pull request merged**